src/main/java/com/marklogic/mapreduce/ContentWriter.java

```diff
@@ -544,6 +540,18 @@ protected void insertBatch(Content[] batch, int id)
                     // not retryable so trying to connect to the replica
                     LOG.warn(getFormattedBatchId() +
                         "RequestServerException: " + e.getMessage());
+                } else if (e instanceof ServerConnectionException) {
+                    // Bug:55607 - When the # of requests per connection reaches
+                    // the limit of Azure application gateway, inform the user
+                    // properly and stop retrying.
+                    if (e.getMessage().contains(
+                        "Server issued connection close within a batch")) {
```

Comment:
My only feedback is that if someone changes the message in xcc this would stop working. I wish we had error codes in xcc (but that's not a big deal). We're OK since you're maintaining both mlcp and xcc. Something to think about in the future when building similar tools.

Reply:
I totally agree with your point. Will enhance this and make it more robust when implementing the enhancement.